### ISSUE TITLE
Feedback: test compile errors could give more useful info

### ISSUE THREAD

**Issue by johanlunds** [DATA-DOG/godog#24]
Example:

```
$ godog .
# command-line-arguments
/tmp/1461229988472036489godog.go:70: no new variables on left side of :=
$
```

Would be nicer with:

```
my_test.go:50: no new variables on left side of :=
```

Perhaps check how `go test` does it.

**Comment by l3pp4rd** [comment]
yes, that would be a good enhancement

**Comment by l3pp4rd** [comment]
it is sad that go does not expose the build tools. in order to build a package with _test.go files is a little too complicated. [gb](https://github.com/constabulary/gb) has replicated the **go** build tool as a library, but I would not like to have it as a dependency. (thinking about having all dependencies inlined in the same package). It could be tried though.

To prevent such errors, it probably would make sense to test the package against errors while building and print the correct error message instead. Also, it could keep the generated go file when the exit code is different. Probably this should be resolved first, while looking for more sophisticated options like build tooling.

**Comment by l3pp4rd** [comment]
while giving another shot at go 1.4 **TestMain** function, was able to produce wanted behavior. This seems to be more pragmatic **go** approach in handling similar requirements - running a custom test suite with standard go build toolchain. Though, go versions until 1.4 won't be supported, which is not so dramatic. The branch is still in progress, but should be available in a week or two for release 0.4.
